parser.py

```diff
@@ -1 +1,40 @@
-# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
\ No newline at end of file
+# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
+from bs4 import BeautifulSoup as bs
+from bs4 import SoupStrainer as ss
+import re
+import urllib2 as ul
+ 	
+def links(webpage):
+	links = []
+	biglist = ul.urlopen(webpage)
+	soup = bs(biglist, "html.parser")
+	tablebody = soup.body.find(id="content")
+	bodycontent = tablebody.find(id="bodyContent")
+	table = bodycontent.find(id="mw-content-text").table
+	for n in range(1, 505):
+		row = table.find_all('tr')[n]
+		name = row.find_all('td')[1]#this is to get the security td's 
+		for element in name:
+			links.append('http://en.wikipedia.org/'+element.get('href'))
+	print links
```

Comment:
Let's not print. Instead, let's return the things we care about: names and links. You can return two things:
```python
return names, links
```